subsys/bluetooth/gatt_dm.c

```diff
@@ -558,7 +558,7 @@ const struct bt_gatt_dm_attr *bt_gatt_dm_attr_next(
 	const struct bt_gatt_dm_attr *prev)
 {
 	if (!prev) {
-		prev = dm->attrs;
```

Comment:
From the function documentation: `@param[in] prev Previous attribute or NULL if we wish to get first attribute (just after service).`.

Returning `dm->attrs` you will return service attribute and that's not intention of this function. Can you verify behavior of this function. Maybe the "print" function should be fixed instead? We are not using this function anywhere but "print". It's a public API function, so changes in behavior should be made with caution.